### ISSUE TITLE
safeCopy is empty when using stPipe, stSelect doesn't find selected row

### ISSUE THREAD

**Issue by lucaalex87** [lorenzofox3/Smart-Table#398]
Hi,

**TL;DR** - stPipe makes select function not work, because safeCopy is empty (data loaded via custom pipe, select doesn't have any rows to work with). I think safeCopy should be updated after the custom pipe returns, but cannot find where to call (**updated** _code below_ ) updateSafeCopy method.

I ran into this issue that when using the custom pipe (which works fine, all data loaded from server and so on), I also tried to add the row selection. But it didn't work, so I investigated and I think I have a fix, but I'm not sure it's the best one.

I believe the issue lies with the safeCopy array being empty, due to it being initialized when the st-table defined array is empty (data not yet loaded from custom pipe).

Then in the select method, it uses safeCopy to find the row that is being selected.

I think the updateSafeCopy method needs to be used even when stSafeSrc is not used (but when st-pipe is), and to copy from displayGetter even after the initialization (first few lines).

```
if ($attrs.stSafeSrc) {
   //copy for when stSafeSrc is used
   safeCopy = copyRefs(safeGetter($scope));
} else {
   //copy from display getter (for when custom pipe is used)
   safeCopy = copyRefs(displayGetter($scope))
}
```

And for it to work, I also called updateSafeCopy inside the select method, first line. I do realize it's not optimal, and was looking for a place to call updateSafeCopy after the custom pipe method returns. I don't have a reference to the table controller in the custom pipe, so I'm not sure where to call updateSafeCopy, so that it's doesn't run at every row select.

It does work like this, but I feel that it does more work than needed.

What do you think?
